Mathlib/RingTheory/FractionalIdeal/Extended.lean

```diff
@@ -209,6 +209,52 @@ theorem coe_extendedHomₐ_eq_span (I : FractionalIdeal A⁰ K) :
     IsLocalization.algebraMap_eq_map_map_submonoid A⁰ B K L]
   rfl
 
+variable [IsIntegrallyClosed A] [IsIntegrallyClosed B]
+
+theorem le_one_of_extendedHomₐ_le_one (hI : extendedHomₐ L B I ≤ 1) : I ≤ 1 := by
+  contrapose! hI
+  rw [SetLike.not_le_iff_exists] at hI ⊢
+  obtain ⟨x, hx₁, hx₂⟩ := hI
+  refine ⟨algebraMap K L x, ?_, ?_⟩
+  · simpa [← FractionalIdeal.mem_coe, IsLocalization.algebraMap_eq_map_map_submonoid A⁰ B K L]
+      using subset_span <| Set.mem_image_of_mem _ hx₁
+  · contrapose! hx₂
+    rw [mem_one_iff, ← IsIntegrallyClosed.isIntegral_iff] at hx₂ ⊢
+    exact IsIntegral.tower_bot_of_field <| isIntegral_trans _ hx₂
+
+theorem extendedHomₐ_le_one_iff : extendedHomₐ L B I ≤ 1 ↔ I ≤ 1 :=
+  ⟨fun h ↦ le_one_of_extendedHomₐ_le_one L B h, fun a ↦ extended_le_one_of_le_one L _ I a⟩
+
+variable [IsDedekindDomain A] [IsDedekindDomain B]
+
+theorem one_le_extendedHomₐ_iff (hI : I ≠ 0) : 1 ≤ extendedHomₐ L B I ↔ 1 ≤ I := by
+  rw [← inv_le_inv_iff ((extendedHomₐ_eq_zero_iff _ _).not.mpr hI) (by simp), inv_one, ← map_inv₀,
+    extendedHomₐ_le_one_iff, inv_le_comm hI (by simp), inv_one]
+
+theorem extendedHomₐ_eq_one_iff (hI : I ≠ 0) : extendedHomₐ L B I = 1 ↔ I = 1 := by
+  rw [le_antisymm_iff, extendedHomₐ_le_one_iff, one_le_extendedHomₐ_iff _ _ hI, ← le_antisymm_iff]
+
+variable (A K) in
+theorem extendedHomₐ_injective :
+    Function.Injective (fun I : FractionalIdeal A⁰ K ↦ extendedHomₐ L B I) := by
+  intro I J h
+  dsimp only at h
+  by_cases hI : I = 0
+  · rwa [hI, map_zero, eq_comm, extendedHomₐ_eq_zero_iff L B, eq_comm, ← hI] at h
+  by_cases hJ : J = 0
+  · rwa [hJ, map_zero, extendedHomₐ_eq_zero_iff L B, ← hJ] at h
+  rwa [← mul_inv_eq_one₀ ((extendedHomₐ_eq_zero_iff _ _).not.mpr hJ), ← map_inv₀, ← map_mul,
+    extendedHomₐ_eq_one_iff _ _ (mul_ne_zero hI (inv_ne_zero hJ)), mul_inv_eq_one₀ hJ] at h
+
+theorem _root_.Ideal.map_algebraMap_injective :
+    Function.Injective (fun I : Ideal A ↦ I.map (algebraMap A B)) := by
```

Comment:
```suggestion
    Function.Injective (Ideal.map (algebraMap A B)) := by
```
Does this not work? (not tested)

Reply:
And the morally correct proof is probably `integral => faithfully flat => comap (map I) = I` and you can probably get rid of the Dedekind domain assumption upstairs. 

(See `Module.Flat.instOfIsDedekindDomainOfNoZeroSMulDivisors` and `Module.FaithfullyFlat.of_specComap_surjective` and `Ideal.comap_map_eq_self_of_faithfullyFlat` and `Ideal.exists_ideal_over_prime_of_isIntegral_of_isDomain`)

Reply:
Indeed, your proof is better. In this case, I suggest that I close this PR and reopen another one with the better proof.

Reply:
But `FractionalIdeal` is probably not used outside of the context of Dedekind domains so this is generalization is more useless and I am happy to see this PR merged first.